apps/OpenSpace/ext/launcher/src/profile/scriptlogdialog.cpp

```diff
@@ -0,0 +1,102 @@
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include "profile/scriptlogdialog.h"
+#include <ghoul/filesystem/filesystem.h>
+#include <openspace/engine/configuration.h>
+
+#include "profile/line.h"
+#include <openspace/scene/profile.h>
+#include <QDialogButtonBox>
+#include <QLabel>
+#include <QFile>
+#include <QTextStream>
+#include <QVBoxLayout>
+
+ScriptlogDialog::ScriptlogDialog(KeybindingsDialog* bindingDialog,
+                                                 QWidget* parent)
+    : QDialog(parent)
+    , _bindingDialog(bindingDialog)
+{
+    setWindowTitle("Scriptlog");
+    createWidgets();
+ 
+    QFile file(QString::fromStdString(absPath("${LOGS}/scriptLog.txt")));
+    if (file.open(QIODevice::ReadOnly | QIODevice::Text)) {
+        QTextStream in(&file);
+        while (!in.atEnd()) {
+            QString line = in.readLine();
+            //removing return from a few statments
+            //these are usually generated by gui panels
+            line.remove(QRegularExpression("^return "));
+            if (line.length() > 0) {
+                _scriptlogList->addItem(line);
+            }
+        }
+        file.close();
+    }
+}
+
+void ScriptlogDialog::createWidgets() {
+    QBoxLayout* layout = new QVBoxLayout(this);
+    {
+        QLabel* heading = new QLabel("Choose commands from log/scriptLog.txt");
+        heading->setObjectName("heading");
+        layout->addWidget(heading);
+    }
+
+    _scriptlogList = new QListWidget;
+    _scriptlogList->setSelectionMode(QAbstractItemView::SelectionMode::MultiSelection);
+    layout->addWidget(_scriptlogList);
+    
+    layout->addWidget(new Line);
+    {
+        QDialogButtonBox* buttons = new QDialogButtonBox;
+        buttons->setStandardButtons(QDialogButtonBox::Save | QDialogButtonBox::Cancel);
+        connect(
+            buttons, &QDialogButtonBox::accepted,
+            this, &ScriptlogDialog::saveChosenScripts
+        );
+        connect(
+            buttons, &QDialogButtonBox::rejected,
+            this, &ScriptlogDialog::reject
+        );
+        layout->addWidget(buttons);
+    }
+}
+
+void ScriptlogDialog::saveChosenScripts() {
+    std::string chosenScripts = "";
```

Comment:
No need to initialize the string to an empty string as the default constructor already is guaranteed to do that